src/callbacks_stage/positivity_adaptive_filter.jl

```diff
@@ -0,0 +1,60 @@
+# By default, Julia/LLVM does not use fused multiply-add operations (FMAs).
+# Since these FMAs can increase the performance of many numerical algorithms,
+# we need to opt-in explicitly.
+# See https://ranocha.de/blog/Optimizing_EC_Trixi for further details.
+@muladd begin
+
+struct FilterParam{FactorType<:Real}
+  relaxation_factor_entropy_var::FactorType
+  relaxation_factor_cons_var   ::FactorType
+end
+
+abstract type AdaptiveFilter end
+struct SecondOrderExponentialAdaptiveFilter{FactorType<:Real} <: AdaptiveFilter
+  param::FilterParam{FactorType}
+end
+struct ZhangShuScalingAdaptiveFilter{FactorType<:Real} <: AdaptiveFilter
+  param::FilterParam{FactorType}
+end
+
+function SecondOrderExponentialAdaptiveFilter(; relaxation_factor_entropy_var,
+                                                relaxation_factor_cons_var)
+  SecondOrderExponentialAdaptiveFilter(FilterParam(relaxation_factor_entropy_var,
+                                                   relaxation_factor_cons_var))
+end
+
+function ZhangShuScalingAdaptiveFilter(; relaxation_factor_entropy_var,
+                                         relaxation_factor_cons_var)
+  ZhangShuScalingAdaptiveFilter(FilterParam(relaxation_factor_entropy_var,
+                                            relaxation_factor_cons_var))
+end
+
+function get_relaxation_factor_cons_var(filter::AdaptiveFilter)
+  return filter.param.relaxation_factor_cons_var
+end
+
+function get_relaxation_factor_entropy_var(filter::AdaptiveFilter)
+  return filter.param.relaxation_factor_entropy_var
+end
+
+struct CompressibleFlowBound{RhoMin, RhoMax, RhoeMin, RhoeMax, VlastMax}
```

Comment:
This is not very general and clean, but serves the purpose (for Euler and CNS)... How do you feel about this?